### PR TITLE
Fix failing UserId test

### PR DESCRIPTION
**Pull request by tasdomas**
Also added a few minor testify usage cleanups.
Isolated UserId test to not polute XDG_CONFIG_HOME

### CODE REVIEW

File: iterative/utils/analytics_test.go [hunk 1]
```diff
@@ -36,18 +38,20 @@ func TestUserId(t *testing.T) {
 		"user_id": userId,
 	}
 	json, _ := json.MarshalIndent(data, "", " ")
+	err := os.MkdirAll(filepath.Dir(old), 0755)
+	assert.Nil(t, err)
+	err = ioutil.WriteFile(old, json, 0644)
+	assert.Nil(t, err)
```
Comment (0x2b3bfa0) [inline iterative/utils/analytics_test.go:44]:
```suggestion
	assert.NoError(t, err)
	err = ioutil.WriteFile(old, json, 0644)
	assert.NoError(t, err)
```

File: iterative/utils/analytics_test.go [hunk 2]
```diff
@@ -36,18 +38,20 @@ func TestUserId(t *testing.T) {
...
 
-	_ = os.MkdirAll(filepath.Dir(old), 0644)
-	_ = ioutil.WriteFile(old, json, 0644)
-
-	id, _ := UserId()
-	assert.Equal(t, len(id) == 36, true)
+	id, err := UserId()
+	assert.Nil(t, err)
```
Comment (0x2b3bfa0) [inline iterative/utils/analytics_test.go:47]:
```suggestion
	assert.NoError(t, err)
```

**Review by 0x2b3bfa0 — COMMENTED** [review]
[`asssert.NoError`](https://pkg.go.dev/github.com/stretchr/testify/assert#NoError) is a more explicit alternative

File: iterative/utils/analytics_test.go (hunk 1 above)
Comment (tasdomas) [reply iterative/utils/analytics_test.go]: Good catch, thank you!

Comment (0x2b3bfa0) [reply iterative/utils/analytics_test.go]: Also, if you like it, the check could be performed in the same line, e.g.

```go
assert.NoError(t, ioutil.WriteFile(old, json, 0644))
```

File: iterative/utils/analytics_test.go [hunk 3]
```diff
@@ -36,18 +38,20 @@ func TestUserId(t *testing.T) {
...
+	assert.Len(t, id, 36)
 
 	if !IsCI() {
-		assert.Equal(t, userId == id, true)
+		assert.Equal(t, userId, id)
 
 		_, err := os.Stat(new)
-		assert.Equal(t, !os.IsNotExist(err), true)
+		assert.True(t, !os.IsNotExist(err))
```
[the remark this answers is not in the document]
Comment (tasdomas) [reply iterative/utils/analytics_test.go:53]: :+1:

[the remark this answers is not in the document]
Comment (0x2b3bfa0) [reply iterative/utils/analytics_test.go]: Resolved with https://github.com/iterative/terraform-provider-iterative/pull/636/commits/7367d8a0820da8f164199656c7c9087b550a1d8e

**Review by 0x2b3bfa0 — APPROVED** [review]
:shipit:

**Review by dacbd — APPROVED** [review]

**Pull request merged**